lib/std/os/linux.zig

```diff
@@ -121,6 +121,28 @@ pub fn utimensat(dirfd: i32, path: ?[*:0]const u8, times: *const [2]timespec, fl
     return syscall4(.utimensat, @bitCast(usize, @as(isize, dirfd)), @ptrToInt(path), @ptrToInt(times), flags);
 }
 
+pub fn fallocate(fd: i32, mode: i32, offset: u64, len: u64) usize {
+    if (@sizeOf(usize) == 4) {
+        return syscall6(
+            .fallocate,
+            @bitCast(usize, @as(isize, fd)),
+            @bitCast(usize, @as(isize, mode)),
+            @truncate(usize, offset >> 32),
```

Comment:
Pro tip: try running this test with `zig test -target i386-linux ../lib/std/os/linux/test.zig`, you swapped the upper/lower halves.

Reply:
Thanks for the tip! Can we always assume little-endian kernel for `syscall6` all 32-bit targets?